### PR TITLE
Add gltf example

### PR DESCRIPTION
**Pull request by alteous**
This is in preparation for glTF support. The difference between the `gltf` example and the `pbr` example is that the former handles node transforms.

### CODE REVIEW

File: examples/gltf.rs [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+    let head = gltf.nodes().nth(3).unwrap();
+    // Meshes must persist for the lifetime of the rendered item.
+    let mut meshes = VecMap::<three::Mesh>::with_capacity(gltf.meshes().len());
+    // Groups *must* be created before meshes.
+    let root = make_group(head, &buffers, &mut meshes, &mut win.factory);
+
+    win.scene.add(&root);
+
+    while win.update() && !three::KEY_ESCAPE.is_hit(&win.input) {
+        if three::Button::Key(three::Key::Q).is_hit(&win.input) {
```
Comment (alteous) [inline examples/gltf.rs]: @shmutalov This is the first person camera implementation with the keyboard.

**Comment by alteous** [comment]
I'm having trouble rebasing these commits into one.

**Comment by vitvakatu** [comment]
@alteous try to rebase in few commits?

**Comment by alteous** [comment]
Yeah, I agree. It's mainly to update to `gltf` version 0.9, but I could do that in another more specific PR.

**Pull request closed without merging**